src/lib/private_components/layered-map-resources/DrawControls.react.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { withLeaflet } from 'react-leaflet';
+import  {EditControl}  from 'react-leaflet-draw';
+import "leaflet-draw/dist/leaflet.draw.css";
+import L from 'leaflet';
+
+
+// work around broken icons when using webpack, see https://github.com/PaulLeCam/react-leaflet/issues/255
+
+delete L.Icon.Default.prototype._getIconUrl;
+L.Icon.Default.mergeOptions({
+  iconRetinaUrl: require('leaflet/dist/images/marker-icon-2x.png'),
+  iconUrl: require('leaflet/dist/images/marker-icon.png'),
+  shadowUrl: require('leaflet/dist/images/marker-shadow.png'),
+})
+
+/**  Leaflet-draw:edit event does not return marker type.
+*    Helper function to find marker type.
+*    https://stackoverflow.com/questions/18014907/leaflet-draw-retrieve-layer-type-on-drawedited-event
+**/
+var getShapeType = (layer) => {
+
+    switch(true) {
+        case (layer instanceof L.Rectangle): {
+            return 'rectangle'
+        }
+        case (layer instanceof L.Circle): {
+            return 'circle'
+        }
+        case (layer instanceof L.Marker): {
+            return 'marker'
+        }
+        case (layer instanceof L.Polygon): {
+            return 'polygon'
+        }
+        case (layer instanceof L.Polyline): {
+            return 'polyline'
+        }
+        default: {
+            return null
+        }
+    }
+}
+
+class DrawControls extends Component {
+
+    _onEdited(e) {
+        e.layers.eachLayer( (layer) => {
+            const layertype = getShapeType(layer)
+            if (layertype === 'polyline') {
+                const coords = layer._latlngs.map(p => {
+                  return [p.lat, p.lng]
+                })
+                this.props.lineCoords(coords)
+            }
+            if (layertype === 'marker') {
+                this.props.markerCoords([layer._latlng.lat, layer._latlng.lng])  
+            }
+        })
+    }
+  
+    removeLayers(layertype) {
+        var {edit} = this.refs
+        var layerContainer = edit.leafletElement.options.edit.featureGroup
+        var layers = layerContainer._layers
+        var layer_ids = Object.keys(layers)
+        let layer
+        for (var i = 0; i < layer_ids.length-1; i++) {
+            layer = layers[layer_ids[i]]
+            if (getShapeType(layer) === layertype) {
+                layerContainer.removeLayer(layer._leaflet_id)
+            }
+        }
+    }
```

Comment:
Remove all `var` in favour of `const` and `let`? 

Can `let layer` be removed and instead add `let` in front of `layer` on line 70? (not tried)